draft-ietf-dnsop-svcb-https.md

```diff
@@ -1082,18 +1082,58 @@ single-client granularity.
 
 ## Interaction with Alt-Svc
 
-Clients that do not implement support for Encrypted ClientHello MAY
-skip the HTTPS RR query
-if a usable Alt-Svc value is available in the local cache.
-If Alt-Svc connection fails, these clients SHOULD fall back to the HTTPS RR
-client connection procedure ({{client-behavior}}).
-
-Clients that implement support for ECH MUST perform the HTTPS RR query first,
-and MUST only make use of Alt-Svc when operating in SVCB-optional mode (see
-{{ech-client-behavior}}).
-
-This specification does not alter the DNS records used when connecting
-to an Alt-Svc hostname (typically A and/or AAAA only).
+Clients that implement support for both Alt-Svc and HTTPS records SHOULD
+retrieve any HTTPS records for the Alt-Svc alt-authority, and ensure that
+their connection attempts are consistent with both the Alt-Svc parameters
+and any received HTTPS SvcParams.  If present, the HTTPS record's TargetName
+and port override the alt-authority.  For example, suppose that
+"https://example.com" sends an Alt-Svc field value of:
+
+~~~ HTTP
+Alt-Svc: h2="alt.example:443", h2="alt2.example:443", h3=":8443"
+~~~
+
+The client would retrieve the following HTTPS records:
+
+    alt.example.              IN HTTPS 1 . alpn=h2,h3 ech=...
+    alt2.example.             IN HTTPS 1 alt2b.example. alpn=h3 ech=...
+    _8443._https.example.com. IN HTTPS 1 alt3.example. (
+        port=9443 alpn=h2,h3 ech=... )
+
+Based on these inputs, the following connection attempts would always be
+allowed:
+
+* HTTPS over TLS to `alt.example:443` (Alt-Svc and HTTPS record)
```

Comment:
"HTTPS over TLS" -> "HTTP/2 over TLS"?

Also, I don't understand what "(Alt-Svc and HTTPS record)" means here.  Is it "(Consistent with both Alt-Svc and HTTPS record)"? Otherwise, I start wondering if this is trying to say something around SVCB vs direct connection.

Is it that

Reply:
It says "HTTPS over TLS" because this is about what connections the client may attempt, not about what ALPN the server will choose on that connection.

> Is it "(Consistent with both Alt-Svc and HTTPS record)"?

Yes.  I've changed the text to that phrasing.

Reply:
> It says "HTTPS over TLS" because this is about what connections the client may attempt, not about what ALPN the server will choose on that connection.

Good point, but in that case, shouldn't it be "TLS over TCP"? I believe that would be the more accurate description of the connection itself and is more consistent with the couple other places in the draft that refer to the connection type.

Reply:
(Or maybe "HTTPS over TCP"?)

Reply:
OK, changed to "HTTPS over TCP".